DragaliaAPI/Controllers/Dragalia/CharaController.cs

```diff
@@ -420,35 +420,43 @@ public async Task<DragaliaResult> CharaLimitBreak([FromBody] CharaLimitBreakRequ
         DbPlayerCharaData playerCharaData = await this.unitRepository
             .GetAllCharaData(this.DeviceAccountId)
             .FirstAsync(chara => chara.CharaId == (Charas)request.chara_id);
+
         CharaData charaData = MasterAsset.CharaData.Get(playerCharaData.CharaId);
+
         playerCharaData.Rarity = 5;
-        bool hasSpiral = charaData.MaxLimitBreakCount > 4;
         playerCharaData.Level = (byte)(
-            CharaConstants.MaxLevel + (hasSpiral ? CharaConstants.AddMaxLevel : 0)
+            CharaConstants.MaxLevel + (charaData.HasManaSpiral ? CharaConstants.AddMaxLevel : 0)
         );
         playerCharaData.Exp = CharaConstants.XpLimits[playerCharaData.Level - 1];
-        playerCharaData.HpBase = hasSpiral ? (ushort)charaData.AddMaxHp1 : (ushort)charaData.MaxHp;
-        playerCharaData.AttackBase = hasSpiral
+        playerCharaData.HpBase = charaData.HasManaSpiral
+            ? (ushort)charaData.AddMaxHp1
+            : (ushort)charaData.MaxHp;
+        playerCharaData.AttackBase = charaData.HasManaSpiral
             ? (ushort)charaData.AddMaxAtk1
             : (ushort)charaData.MaxAtk;
-        if (playerCharaData.ManaNodeUnlockCount < 50)
-        {
-            playerCharaData.HpNode += (ushort)charaData.McFullBonusHp5;
-            playerCharaData.AttackNode += (ushort)charaData.McFullBonusAtk5;
-        }
         playerCharaData.LimitBreakCount = (byte)charaData.MaxLimitBreakCount;
-        ManaNodes maxManaNodes = hasSpiral ? ManaNodes.Circle7 : ManaNodesUtil.MaxManaNodes;
+
+        IEnumerable<int> maxManaNodes = ManaNodesUtil.GetSetFromManaNodes(
+            charaData.HasManaSpiral ? ManaNodes.Circle7 : ManaNodesUtil.MaxManaNodes
+        );
         Dictionary<CurrencyTypes, int> usedCurrencies = new();
         Dictionary<Materials, int> usedMaterials = new();
         HashSet<int> unlockedStories = new();
+
         await CharaManaNodeUnlock(
-            ManaNodesUtil.GetSetFromManaNodes(maxManaNodes),
+            maxManaNodes,
             playerCharaData,
             usedCurrencies,
             usedMaterials,
             unlockedStories,
             CharaUpgradeMaterialTypes.Omnicite
         );
+
+        if (charaData.HasManaSpiral)
+        {
+            playerCharaData.AdditionalMaxLevel = 20;
```

Comment:
isn't this already modified by CharaManaNodeUnlock?

Reply:
There isn't really a vanilla way to upgrade both pre-spiral and post-spiral nodes at the same time except for when using omnicite, and CharaManaNodeUnlock doesn't add an additional 5 levels for limit breaking to spiral since it's only in charge of unlocking nodes not limit breaking
I could change it to add 5 to AdditionalMaxLevel in the method if it detects a new 50 mc unlock + spiralled nodes but, idk it made more sense to me to just set it in omnicite upgrading

Reply:
Ah sorry I misread and thought this was += 20 and it would be increased twice. This is fine